packages/astro/src/@types/astro.ts

```diff
@@ -2163,6 +2163,34 @@ export interface AstroUserConfig {
 			 */
 			schema?: EnvSchema;
 		};
+
+		/**
+		 * @docs
+		 * @name experimental.serverIslands
+		 * @type {boolean}
+		 * @default `false`
+		 * @version 4.12.0
+		 * @description
+		 *
+		 * Enables Server Islands, the ability to defer a component to render asynchronously after the page has already rendered.
+		 *
+		 * ```js
+		 * {
+		 *   experimental: {
+		 *     serverIslands: true,
+		 *   },
+		 * }
+		 * ```
+		 * 
+		 * Use the `server:defer` directive on any component, Astro or framework, and the component will not render initially.
+		 *
+		 * ```astro "server:defer"
+		 * <Avatar server:defer />
```

Comment:
What happens if the directive is used without the flag? Will there be an error saying users should enable it?